src/web/builds.rs

```diff
@@ -37,11 +38,30 @@ impl_webpage! {
 pub fn build_list_handler(req: &mut Request) -> IronResult<Response> {
     let router = extension!(req, Router);
     let name = cexpect!(req, router.find("name"));
-    let version = cexpect!(req, router.find("version"));
+    let req_version = router.find("version");
```

Comment:
I'm confused: Where does `find` come from? I don't see it on https://docs.rs/router/0.6.0/router/struct.Router.html. I do see a find() method on https://docs.rs/router/0.6.0/router/struct.Params.html#method.find, is that it?

Reply:
Yeah, this is a bit of magic introduced by Iron's router middleware (accessed via the `extension!` macro). `router::Params` is actually a re-exported [`route_recognizer::Params`](https://docs.rs/route-recognizer/0.3.0/route_recognizer/struct.Params.html).

Reply:
Specifically it's [this `impl Key for Router { type Value = Params; }`](https://docs.rs/router/0.6.0/router/struct.Router.html#impl-Key) that means `extension!(req, Router): Params`..